homeassistant/components/homekit/type_switches.py

```diff
@@ -1,25 +1,61 @@
 """Class to hold all switch accessories."""
 import logging
 
-from pyhap.const import CATEGORY_SWITCH
+from pyhap.const import CATEGORY_OUTLET, CATEGORY_SWITCH
 
+from homeassistant.components.switch import DOMAIN as SWITCH
 from homeassistant.const import (
     ATTR_ENTITY_ID, SERVICE_TURN_ON, SERVICE_TURN_OFF, STATE_ON)
 from homeassistant.core import split_entity_id
 
 from . import TYPES
 from .accessories import HomeAccessory
-from .const import SERV_SWITCH, CHAR_ON
+from .const import CHAR_ON, CHAR_OUTLET_IN_USE, SERV_OUTLET, SERV_SWITCH
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@TYPES.register('Outlet')
+class Outlet(HomeAccessory):
+    """Generate an Outlet accessory for a switch entity."""
+
+    def __init__(self, *args):
+        """Initialize an Outlet accessory object."""
+        super().__init__(*args, category=CATEGORY_OUTLET)
+        self.flag_target_state = False
+
+        serv_outlet = self.add_preload_service(SERV_OUTLET)
+        self.char_on = serv_outlet.configure_char(
+            CHAR_ON, value=False, setter_callback=self.set_state)
+        self.char_outlet_in_use = serv_outlet.configure_char(
+            CHAR_OUTLET_IN_USE, value=False)
```

Comment:
I'm uncertain if we not just should set the default to `True` and leave it at that, don't change it later.

Reply:
`OutletInUse` is supposed to represent if a device is physically plugged into the outlet, regardless if it's powered on or not. Since we don't know that information from HASS, I chose to follow the power state and assume nothing is plugged into the outlet if the power state is off.

I would prefer to keep this current method because I think it's more intuitive for users who aren't familiar with the individual HomeKit characteristics. If we set outlet in use default to `true`, then maybe users will get confused (ex. power state says `off` but outlet in use says `true`).

Reply:
I don't think your assumption here is correct. At least the way I see it a switch just turns something on or off that is already plugged in. Nothing is removed in between. Changing `OutletInUse` would therefore be more misleading IMO, since someone might assume that it's the correct state if it's changing.